Frameworks/Foundation/NSOperation.mm

```diff
@@ -131,139 +128,109 @@ - (void)setThreadPriority:(double)priority {
  @Status Interoperable
 */
 - (void)setCompletionBlock:(void (^)(void))block {
-    id oldBlock = priv->completionBlock;
-    priv->completionBlock = [block copy];
-    [oldBlock release];
+    pthread_mutex_lock(&_completionBlockLock);
+    [_completionBlock autorelease];
```

Comment:
If you set op.completionBlock = op.completionBlock, you will likely destroy the block here./

---
In reply to: [69352043](https://github.com/Microsoft/WinObjC/pull/602#discussion_r69352043) [](ancestors = 69352043,69351630)